### PR TITLE
When selected path is being looked for menu by comparing string-based url It should select most nested url

### PR DESCRIPTION
**Pull request by neTp9c**
Marking menu item as active is not always right.

It's wrong for situation when first element of second menu level and element of first menu level links to same url:
  * homepage (~/)
  * about us (~/about-us)
    * about as (~/about-us)
    * our news (~/about-us/our-news)
      * news 1 (~/about-us/our-news/news-1)
      * news 2 (~/about-us/our-news/news-2)
  * contacts

News 1 and News 2 are not actually added to menu but have respective url.

When current (routed) page is "news 1" then "about as" page from second menu level is marked active (instead of "our news" page)

This pull request solves that problem.

### CODE REVIEW

**Comment by sebastienros** [comment]
Please remove the reformat, we can't see what the actual changes are.

**Comment by neTp9c** [comment]
It was not a simple reformat. I split one function into three (it's more like refactoring).
I do it because finding active menu item by url-based string and route values looks like independent methods for me (current method has too many if-else clauses).
Anyway I joined they together just as it was before.

**Comment by sebastienros** [comment]
It needs sign-off and also testing the other part of the code to ensure it's not breaking other scenarios.

**Comment by neTp9c** [comment]
I added sign-off-by.
I don't know what I can do about testing. If you give me direction I will test it.
But I can't imagine any right scenario that could be broken by this code. Picks the menu item with the most nested URL (when using URL comparison) looks best choice for me.

If some one ordered his menu items in a reverse order:
- level-1/level-2
    - level-1
and current page is level-1/level-2/level-3 that in this case only level-1/level-2 item will be marked as active.

**Comment by sebastienros** [comment]
> I can't imagine any right scenario that could be broken by this code

This is one of the funniest comments ever ;) 

> I don't know what I can do about testing

The sign-off label is for someone else than the creator of the PR to test the feature. It's always good to have someone else look at it, because even the most simple changes might break something.

File: src/Orchard/UI/Navigation/NavigationHelper.cs [hunk 1]
```diff
@@ -96,12 +102,17 @@ public static class NavigationHelper {
         /// <param name="compareUrls">Should compare raw string URLs instead of route data.</param>
         /// <returns>A stack with the selection path being the last node the currently selected one.</returns>
         private static Stack<MenuItem> SetSelectedPath(IEnumerable<MenuItem> menuItems, HttpRequestBase currentRequest, RouteValueDictionary currentRouteData, bool compareUrls) {
+            var selectedPathes = new List<Stack<MenuItem>>();
```
Comment (rtpHarry) [inline src/Orchard/UI/Navigation/NavigationHelper.cs]: typo, should just be `selectedPaths`

**Comment by neTp9c** [comment]
:)
I fixed the typo.

**Comment by sfmskywalker** [comment]
I'm running into an issue that this PR seems to fix, so I'll take it for a test drive.

**Comment by sfmskywalker** [comment]
I just tested it and it works nicely! The only thing I would change is rename the private `SetSelectedPath` to `GetSelectedPath`, since it's no longer setting `menuItem.Selected = true`; Instead, the public `SetSelectedPath` is doing that now.

GTG

**Pull request merged**